website/docs/r/kubernetes_cluster_node_pool.html.markdown

```diff
@@ -126,7 +126,9 @@ The following arguments are supported:
 
 * `pod_subnet_id` - (Optional) The ID of the Subnet where the pods in the Node Pool should exist. Changing this forces a new resource to be created.
 
-* `os_sku` - (Optional) Specifies the OS SKU used by the agent pool. Possible values are `AzureLinux`, `Ubuntu`, `Windows2019` and `Windows2022`. If not specified, the default is `Ubuntu` if OSType=Linux or `Windows2019` if OSType=Windows. And the default Windows OSSKU will be changed to `Windows2022` after Windows2019 is deprecated. Changing this forces a new resource to be created.
+* `os_sku` - (Optional) Specifies the OS SKU used by the agent pool. Possible values are `AzureLinux`, `Ubuntu`, `Windows2019` and `Windows2022`. If not specified, the default is `Ubuntu` if OSType=Linux or `Windows2019` if OSType=Windows. And the default Windows OSSKU will be changed to `Windows2022` after Windows2019 is deprecated. Changing this from `AzureLinux` or `Ubuntu` to `AzureLinux` or `Ubuntu` will not replace the resource, otherwise it forces a new resource to be created.
+
+-> **Note:**  This requires that the Preview Feature `Microsoft.ContainerService/OSSKUMigrationPreview` is enabled and the Resource Provider is re-registered, see [the documentation](https://learn.microsoft.com/en-us/azure/azure-linux/tutorial-azure-linux-migration) for more information.
```

Comment:
This should be removed if it's not really a preview feature anymore
```suggestion
```

Reply:
I'll remove it when it's GA, currently we still need this feature registered.

Reply:
Does it though? That flag isn't registered on our testing subscription
```
$ az feature show --namespace "Microsoft.ContainerService" --name "OSSKUMigrationPreview"
{
  "id": "/subscriptions/*/providers/Microsoft.Features/providers/microsoft.ContainerService/features/OSSKUMigrationPreview",
  "name": "microsoft.ContainerService/OSSKUMigrationPreview",
  "properties": {
    "state": "NotRegistered"
  },
  "type": "Microsoft.Features/providers/features"
}
```

And the test passes
![image](https://github.com/hashicorp/terraform-provider-azurerm/assets/25562180/ab7d8210-0a8e-434f-83b9-b2e68af8a6c2)

Reply:
Got it, I'll remove the notice.